### ISSUE TITLE
Decide whether split(..., None) should return (None, None)

### ISSUE THREAD

**Issue by njsmith** [python-trio/exceptiongroup#7]
That's what it does right now. Is there any good reason for this?

`MultiError.filter` passed through `None`, but `MultiError.filter` had a more general system where handlers took exceptions as arguments, and returned new exceptions as return values, and could return `None` to mean "this exception has been successfully handled, no exception replaces it", and that's gone in the new design.

**Comment by WindSoilder** [comment]
Here is my understanding of `handler` in the new picture:
```python
caught, rest = split(.., .., ..)
handle(caught)
```
If `handler` **can't** handle any exceptions  in caught, it will re-raise an exception (whatever user like to raise).  Or we can make sure that `handler` handle exception successfully?

**Comment by njsmith** [comment]
Yeah, I can't think of any cases where we would want to pass `None` into `split`. For `catch`, I guess it does something like:

```python
@contextmanager
def simplified_catch(exc_type, handler):
    try:
        yield
    except BaseException as exc:
        caught, rest = split(exc_type, exc)
        ...
```

So it should never encounter a situation where `exc` is `None`. `caught` or `rest` might be `None`, but `exc` is never `None`.

**Comment by WindSoilder** [comment]
What about introducing type annotation here?
```python
def split(exc_type: Type[BaseException], exc: BaseException, *, match: Optional[Callable] = None) -> Tuple[BaseException, BaseException]:
```
From the defininition, we don't need to check if the `exc` is instance of `Exception`.

**Comment by njsmith** [comment]
Fixed in #10
